pubsub/pubsub_test.go

```diff
@@ -164,19 +204,46 @@ func teardownCB(...string) error {
 	}
 
 	for it := psc.Subscriptions(ctx); true; {
-		t, err := it.Next()
+		s, err := it.Next()
 		if err == iterator.Done {
 			break
 		} else if err != nil {
 			return err
 		}
 
-		if strings.HasPrefix(t.ID(), "bps-unittest-") {
-			if err := t.Delete(ctx); err != nil {
+		if strings.HasPrefix(s.ID(), "bps-unittest-") {
+			if err := s.Delete(ctx); err != nil {
 				return err
 			}
 		}
 	}
 
 	return nil
 }
+
+func seedMessages(topicName string, messages []bps.SubMessage) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	client, err := native.NewClient(ctx, projectID)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	topic, err := client.CreateTopic(ctx, topicName)
+	if err != nil {
+		return err
+	}
+	defer topic.Stop()
+
+	for i, msg := range messages {
+		nativeMsg := &native.Message{
+			Data: msg.Data(),
+		}
+		if _, err := topic.Publish(ctx, nativeMsg).Get(ctx); err != nil {
```

Comment:
@dim IMO, if that works - publish message with only Data set (I've copied this from native pkg example) - we can trim the `PubMessage` struct to just `struct { Data []byte }` or even better - just accept `[]byte` directly.

Also, another observation is that `SubMessage { Data() []byte }` interfaces always goes to just wrapping `[]byte` with `type bps.RawSubMessage []byte`.
I think, we could ditch `SubMessage` interface as well - I don't think we should expose any other methods on it, like Ack/Nack - it hides nicely behind the Subscriber implementation and doing Ack/Nack by `handler.Handle(msg) error`'s result is much better (lib user won't ever forget to Ack/Nack, just return error).

Reply:
@dim please, pay attention to this one as well ^^

Reply:
I'll merge as is, but this still is something to consider.